src/Private/Helpers.ps1

```diff
@@ -3766,4 +3749,212 @@ function Copy-PodeObjectDeepClone {
+
+            # Include milliseconds if they exist and are not excluded
+            if ((($timeSpan.Milliseconds -gt 0) -and !$ExcludeMilliseconds) -or ($output.Count -eq 0)) {
+                $output += "$($timeSpan.Milliseconds)ms"
+            }
+
+            return $output -join ' '
+        }
+
+        'compact' {
+            # Compact format: "dd:hh:mm:ss"
+            $output = '{0:D2}:{1:D2}:{2:D2}:{3:D2}' -f $timeSpan.Days, $timeSpan.Hours, $timeSpan.Minutes, $timeSpan.Seconds
+
+            # Append milliseconds if not excluded
+            if (!$ExcludeMilliseconds) {
+                $output += '.{0:D3}' -f $timeSpan.Milliseconds
+            }
+
+            return $output
+        }
+
+        'verbose' {
+            # Verbose format: "1 day, 2 hours, 3 minutes, 4 seconds"
+            $output = @()
+            if ($timeSpan.Days -gt 0) { $output += "$($timeSpan.Days) day$(if ($timeSpan.Days -ne 1) { 's' })" }
+            if ($timeSpan.Hours -gt 0) { $output += "$($timeSpan.Hours) hour$(if ($timeSpan.Hours -ne 1) { 's' })" }
+            if ($timeSpan.Minutes -gt 0) { $output += "$($timeSpan.Minutes) minute$(if ($timeSpan.Minutes -ne 1) { 's' })" }
+            if ($timeSpan.Seconds -gt 0) { $output += "$($timeSpan.Seconds) second$(if ($timeSpan.Seconds -ne 1) { 's' })" }
+
+            # Include milliseconds if they exist and are not excluded
+            if ((($timeSpan.Milliseconds -gt 0) -and !$ExcludeMilliseconds) -or ($output.Count -eq 0)) {
+                $output += "$($timeSpan.Milliseconds) millisecond$(if ($timeSpan.Milliseconds -ne 1) { 's' })"
+            }
+
+            return $output -join ', '
+        }
+    }
+}
+
+<#
+.SYNOPSIS
+    Checks if the current PowerShell session supports console-like features.
+
+.DESCRIPTION
+    This function determines if the current PowerShell session is running in a host
+    that typically indicates a console-like environment where `Ctrl+C` can interrupt.
+    On Windows, it validates the standard input and output handles.
+    On non-Windows systems, it checks against known supported hosts.
+
+.OUTPUTS
+    [bool]
+    Returns `$true` if running in a console-like environment, `$false` otherwise.
+
+.EXAMPLE
+    Test-PodeHasConsole
+    # Returns `$true` if the session supports console-like behavior.
+#>
+function Test-PodeHasConsole {
+    if (@('ConsoleHost', 'Windows PowerShell ISE Host', 'Visual Studio Code Host') -contains $Host.Name) {
+        if (Test-PodeIsWindows) {
+            $handleTypeMap = @{
+                Input  = -10
+                Output = -11
+                Error  = -12
+            }
+            # On Windows, validate standard input and output handles
+            return [Pode.NativeMethods]::IsHandleValid($handleTypeMap.Input) -and [Pode.NativeMethods]::IsHandleValid($handleTypeMap.Output)
+        }
+        # On Linux or Mac
+        return ([Pode.NativeMethods]::IsTerminal)
```

Comment:
Should this not be `::IsTerminal()` with the `()`, as it's a method not a property?